scipy/optimize/_trustregion_iterative.py

```diff
@@ -0,0 +1,470 @@
+"""Iterative trust-region optimization subproblem."""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.linalg import norm, get_lapack_funcs, solve_triangular, norm
+from ._trustregion import (_minimize_trust_region, BaseQuadraticSubproblem)
+
+__all__ = ['estimate_smallest_singular_value',
+           'singular_leading_submatrix',
+           'IterativeSubproblem']
+
+
+def _minimize_trust_iterative(fun, x0, args=(), jac=None, hess=None,
+                              **trust_region_options):
+    """
+    Minimization of scalar function of one or more variables using
+    a iterative trust-region algorithm.
+
+    Options
+    -------
+    initial_trust_radius : float
+        Initial trust-region radius.
+    max_trust_radius : float
+        Maximum value of the trust-region radius. No steps that are longer
+        than this value will be proposed.
+    eta : float
+        Trust region related acceptance stringency for proposed steps.
+    gtol : float
+        Gradient norm must be less than ``gtol` before successful
+        termination.
+    trust_region_factor, optimization_factor : float
+        Parameters that determines the stop criteria for the subproblem solver.
+        The iterative algoritm for the trust-region subproblem will stop when
+        the computed step ``p`` fullfil
+        ``norm(p)<=trust_region_factor*trust_radius``
+        or when the solution ``m`` for the subproblem fullfil
+        ``m < optimization_factor*m_optimal``, with ``m_optimal`` being the
```

Comment:
OK, only now I started to realize that this condition is really unintuitive. It says that we can achieve the cost function lower than the *optimal (minimal)* cost function. As far as I figured it necessary relates to the trust-region boundary violation. It seems to me that this condition should be removed from the public interface, because it's frankly impossible to understand for a user.

Can you elaborate on that? Didn't I miss anything here?